src/c_wrapper.jl

```diff
@@ -37,35 +37,36 @@ function SCS_solve(T::Union{Type{Direct}, Type{Indirect}},
     n > 0 || throw(ArgumentError("The number of variables in SCSModel must be greater than 0"))
     m > 0 || throw(ArgumentError("The number of constraints in SCSModel must be greater than 0"))
 
-    managed_matrix = ManagedSCSMatrix(m, n, A)
-    matrix = Ref(SCSMatrix(managed_matrix))
-    settings = Ref(SCSSettings(T; options...))
-    data = Ref(SCSData(m, n, Base.unsafe_convert(Ptr{SCSMatrix}, matrix), pointer(b), pointer(c), Base.unsafe_convert(Ptr{SCSSettings},settings)))
-
-    cone = Ref(SCSCone(f, l, q, s, ep, ed, p))
-    info = Ref(SCSInfo())
-
     ws = (:warm_start=>true) in options
 
-    if ws && length(primal_sol) == n && length(dual_sol) == m && length(slack) == m
-        x = primal_sol
-        y = dual_sol
-        s = slack
+    if ws && length(primal_sol) == n && length(dual_sol) == length(slack) == m
+        primal = primal_sol
+        dual = dual_sol
+        slack = slack
```

Comment:
This line can be removed, it has no effect

Reply:
which one?
Do you mean that the default `zeros(...)` for `primal`, `dual` and `slack` should be the default values?

Reply:
how about something like this:
```julia
function SCS_solve(T::Union{Type{Direct}, Type{Indirect}},
        m::Int, n::Int, A::SCSVecOrMatOrSparse, b::Array{Float64},
        c::Array{Float64}, f::Int, l::Int, q::Array{Int}, s::Array{Int},
        ep::Int, ed::Int, p::Array{Float64},
        primal_sol::Vector{Float64}=zeros(n),
        dual_sol::Vector{Float64}=zeros(m),
        slack::Vector{Float64}=zeros(m);
        options...)

    n > 0 || throw(ArgumentError("The number of variables in SCSModel must be greater than 0"))
    m > 0 || throw(ArgumentError("The number of constraints in SCSModel must be greater than 0"))

    ws = (:warm_start=>true) in options
    
    if !(length(primal_sol) == n && length(dual_sol) == length(slack) == m)
        ws && @warn "The provided warmstart doesn't match problem sizes; discarding!"
        primal_sol = zeros(n)
        dual_sol = zeros(m)
        slack = zeros(m)
    else
        if ws # the provided warmstart matches problem sizes and user asked for warmstaring
            nothing
        else 
            primal_sol = fill!(primal_sol, 0.0)
            dual_sol = fill!(dual_sol, 0.0)
            slack = fill!(slack, 0.0)
        end
    end
    
    solution = SCSSolution(pointer(primal_sol), pointer(dual_sol), pointer(slack))
    (...)
    return Solution(primal_sol, dual_sol, slack, info[], status)
end
```

Reply:
I simply mean that `slack = slack` is a no-op